### PR TITLE
also install Python bindings for SHTns 3.5.1

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/s/SHTns/SHTns-3.5.1-foss-2021b.eb [hunk 1]
```diff
@@ -8,14 +8,26 @@ description = """Spherical Harmonic Transform library aimed at high performance
...
 source_urls = ['https://bitbucket.org/nschaeff/shtns/downloads/']
 sources = [SOURCELOWER_TAR_GZ]
 checksums = ['77f8a33b94df8786d2ce9b95cbfbe548f00443625b310b8c64012b22c8a7394f']
 
+dependencies = [
+    ('Python', '3.9.6'),
+    ('SciPy-bundle', '2021.10'),
+]
+
+installopts = " && pip install --no-deps --ignore-installed --prefix %(installdir)s ."
```
Comment (lexming) [inline easybuild/easyconfigs/s/SHTns/SHTns-3.5.1-foss-2021b.eb:22]: What do you think about using an extension instead? Then you get all the goodies of `PythonPackage` and the python module can be searched as extension
```suggestion
# install Python bindings and its dependencies
exts_defaultclass = 'PythonPackage'
exts_default_options = {
    'download_dep_fail': True,
    'use_pip': True,
    'sanity_pip_check': True,
}
exts_list = [
    ('shtns', version, {
        'source_tmpl': SOURCELOWER_TAR_GZ,
        'checksums': ['77f8a33b94df8786d2ce9b95cbfbe548f00443625b310b8c64012b22c8a7394f'],
    }),
]
```

File: easybuild/easyconfigs/s/SHTns/SHTns-3.5.1-foss-2021b.eb [hunk 2]
```diff
@@ -8,14 +8,26 @@ description = """Spherical Harmonic Transform library aimed at high performance
...
+
 sanity_check_paths = {
     'files': ['include/shtns_cuda.h', 'include/shtns.f', 'include/shtns.f03', 'include/shtns.h', 'lib/libshtns.a'],
-    'dirs': [],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
 }
 
+sanity_check_commands = ["python -c 'import shtns'"]
+
```
Comment (lexming) [inline easybuild/easyconfigs/s/SHTns/SHTns-3.5.1-foss-2021b.eb:30]: By using an extension this will be automatic
```suggestion
```

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/s/SHTns/SHTns-3.5.1-foss-2021b.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/s/SHTns/SHTns-3.5.1-foss-2021b.eb]: makes total sense, this is significantly better (though less verbose), thanks @lexming!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3112.skitty.os - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/45848a6c9209684f27f1e33c1a0174aa for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @boegel !

**Pull request merged**